sample/ios/ios/ContentView.swift

```diff
@@ -18,36 +18,36 @@ struct ContentView: View {
     )
 
     var body: some View {
-        
-        return ZStack {
-            if state is LoadFirstPagePaginationState {
-                LoadingIndicatorView()
-            } else if let state = state as? ShowContentPaginationState {
-                GithubReposList(
-                    contentState: state,
-                    dispatchAction: dispatchAction
-                )
-            } else if state is LoadingFirstPageError {
-                // TODO extract standalone widget?
-                Button(action: triggerReloadFirstPage) {
-                    Text("An error has occured.\nClick here to retry.")
-                }
-            }
-        }.onAppear(perform: startStateMachine)
+        contentView()
+            .onAppear(perform: startStateMachine)
 
     }
 
+    @ViewBuilder
+    private func contentView() -> some View {
+        switch state {
+        case is LoadFirstPagePaginationState:
+            LoadingIndicatorView()
+        case let state as ShowContentPaginationState:
+            GithubReposList(contentState: state, dispatchAction: dispatchAction)
+        case is LoadingFirstPageError:
+            ErrorView(action: triggerReloadFirstPage)
+        default:
+            EmptyView()
```

Comment:
Suggestion: I would actually raise an Error here. That would help catch "unhandled" cases in the future instead of "wondering" why things are not displayed. 

But it is your call ...

Reply:
Nice suggestion 👍  -> https://github.com/freeletics/FlowRedux/pull/456/commits/130651fbd07e8e6aace0c5b502ce4be0156f6476